src/hssm/plotting/utils.py

```diff
@@ -79,9 +109,19 @@ def _get_plotting_df(
     pd.DataFrame
         A dataframe with the original data and the extra dimensions.
     """
-    idata_posterior = idata["posterior_predictive"]["rt,response"]
+    if idata is None and data is None:
+        raise ValueError("Either idata or data must be provided.")
+
+    extra_dims = [] if extra_dims is None else extra_dims
+
+    if idata is None:
+        data = _process_data(data, extra_dims)
+
+        data.insert(0, "observed", "observed")
+        return data
 
     # get the posterior samples
+    idata_posterior = idata["posterior_predictive"]["rt,response"]
```

Comment:
seems like logic wise, you could have `idata is None --> True` but then still access `idata` in this line?

Reply:
The function returns if `idata is None` at line 121, so it would not reach here if `idata is None`?

Reply:
ok overlooked the return